### PR TITLE
added UI::SetMaxDoubleClickDist and UI::GetMaxDoubleClickDist

### PR DESCRIPTION
**Pull request by TrevorCash**
Adds a maximum distance parameter to the UI subsystem to enable filtering of double clicks where the first click and second click are far away from each other on the same UIElement.  The distance is defaulted to very large so existing functionality should not be broken.

### CODE REVIEW

**Comment by TrevorCash** [comment]
Fixes #2196

File: Source/Urho3D/UI/UI.h [hunk 1]
```diff
@@ -402,6 +411,10 @@ class URHO3D_API UI : public Object
     Timer clickTimer_;
     /// UI element last clicked for tracking double clicks.
     WeakPtr<UIElement> doubleClickElement_;
+	/// Screen position of first mouse click for double click distance checking.
```
Comment (eugeneko) [inline Source/Urho3D/UI/UI.h]: Tabs to spaces

File: Source/Urho3D/UI/UIEvents.h [hunk 2]
```diff
@@ -56,6 +56,8 @@ URHO3D_EVENT(E_UIMOUSEDOUBLECLICK, UIMouseDoubleClick)
     URHO3D_PARAM(P_ELEMENT, Element);              // UIElement pointer
     URHO3D_PARAM(P_X, X);                          // int
     URHO3D_PARAM(P_Y, Y);                          // int
+	URHO3D_PARAM(P_XBegin, XBegin);                // int
```
Comment (eugeneko) [inline Source/Urho3D/UI/UIEvents.h]: Please use CAPS for parameter names, as everywhere

File: Source/Urho3D/UI/UIEvents.h [hunk 3]
```diff
@@ -56,8 +56,8 @@ URHO3D_EVENT(E_UIMOUSEDOUBLECLICK, UIMouseDoubleClick)
     URHO3D_PARAM(P_ELEMENT, Element);              // UIElement pointer
     URHO3D_PARAM(P_X, X);                          // int
     URHO3D_PARAM(P_Y, Y);                          // int
-	URHO3D_PARAM(P_XBegin, XBegin);                // int
-	URHO3D_PARAM(P_YBegin, YBegin);                // int
+    URHO3D_PARAM(P_XBEGIN, XBEGIN);                // int
```
Comment (eugeneko) [inline Source/Urho3D/UI/UIEvents.h]: Sorry for confusing. I mean `(P_XBEGIN, XBegin)`

File: Source/Urho3D/UI/UI.cpp [hunk 4]
```diff
@@ -95,7 +95,7 @@ UI::UI(Context* context) :
     dragBeginDistance_(DEFAULT_DRAGBEGIN_DISTANCE),
     mouseButtons_(0),
     lastMouseButtons_(0),
-	maxDoubleClickDist_(FLT_MAX),	
+    maxDoubleClickDist_(99999.0f),
```
Comment (eugeneko) [inline Source/Urho3D/UI/UI.cpp]: There is `M_LARGE_VALUE` constant. Is it acceptable?

File: Source/Urho3D/UI/UIEvents.h (hunk 3 above)
Comment (TrevorCash) [reply Source/Urho3D/UI/UIEvents.h]: Ah yes - I see that now.

**Comment by Lumak** [comment]
Is this necessary? Clicks are registered if within the bounds of UIElement, and it's working as designed.  If you need a smaller bound then you'd typically add a smaller bound UIElement, or is this a work around so you don't have to add a smaller bound UIElement?

**Comment by TrevorCash** [comment]
Its helpful. not necessary.  If I want to detect double clicks say, in the root ui element (which is very big) I want to know that I'm clicking in the same general area when a double click is registered.  I find this most helpful when not considering UIElements at all and just wanting metrics on mouse actions in the canvas.

**Comment by Lumak** [comment]
To get mouse position, all you need is call:

`Input::IntVector2 GetMousePosition() const;
`
without having the need to click on anything.

File: Source/Urho3D/UI/UI.cpp (hunk 4 above)
Comment (TrevorCash) [reply Source/Urho3D/UI/UI.cpp]: Awesome! - did not realize that was there - just updated

File: Source/Urho3D/UI/UI.cpp [hunk 5]
```diff
@@ -1635,14 +1642,37 @@ void UI::SendClickEvent(StringHash eventType, UIElement* beginElement, UIElement
...
             endElement->SendEvent(E_CLICKEND, eventData);
-        else if (eventType == E_UIMOUSEDOUBLECLICK)
-            endElement->SendEvent(E_DOUBLECLICK, eventData);
     }
 
     // Send the global event from the UI subsystem last
     SendEvent(eventType, eventData);
 }
 
+void UI::SendDoubleClickEvent(StringHash eventType, UIElement* beginElement, UIElement* endElement, const IntVector2& firstPos, const IntVector2& secondPos, int button,
```
Comment (eugeneko) [inline Source/Urho3D/UI/UI.cpp]: `eventType` parameters seems useless. Could you remove it?

**Comment by eugeneko** [comment]
Well, I'm fine with this PR.
I'll merge it if there is no comments for three days.

**Comment by TrevorCash** [comment]
Cool! I dont plan on making any alterations to this branch.  Thanks Eugene!

Get Outlook for Android<https://aka.ms/ghei36>

File: Source/Urho3D/UI/UI.h [hunk 6]
```diff
@@ -163,7 +165,10 @@ class URHO3D_API UI : public Object
 
     /// Return UI element double click interval in seconds.
     float GetDoubleClickInterval() const { return doubleClickInterval_; }
-
+    
+    /// Get max screen distance for double clicks to register.  default is FLT_MAX.
```
Comment (weitjong) [inline Source/Urho3D/UI/UI.h]: Comment not properly case. The constant value seems to be different than the implementation.

**Pull request merged**